src/Symfony/Component/Routing/Route.php

```diff
@@ -148,14 +148,15 @@ public function getPath()
      * This method implements a fluent interface.
      *
      * @param string $pattern The path pattern
+     * @param bool   $trim    whether or not to trim and format the $pattern.
      *
      * @return Route The current Route instance
      */
-    public function setPath($pattern)
+    public function setPath($pattern, $trim = true)
     {
         // A pattern must start with a slash and must not have multiple slashes at the beginning because the
         // generated path for this route would be confused with a network path, e.g. '//domain.com/path'.
-        $this->path = '/'.ltrim(trim($pattern), '/');
+        $this->path = $trim ? '/'.ltrim(trim($pattern), '/') : $pattern;
```

Comment:
Well, what would you think about doing this instead: https://github.com/symfony/symfony/pull/18702#issuecomment-216617393